intel-sgx/enclave-runner/src/usercalls/mod.rs

```diff
@@ -987,12 +1000,20 @@ impl EnclaveState {
             .expect("Work sender couldn't send data to receiver");
 
         let join_handlers =
-            create_worker_threads(num_of_worker_threads, work_receiver, io_queue_send.clone());
+            create_worker_threads(num_of_worker_threads, work_receiver, io_queue_send.clone(), enclave.exiting.clone());
         // main syscall polling loop
         let main_result =
             EnclaveState::syscall_loop(enclave.clone(), io_queue_receive, io_queue_send, work_sender);
 
         for handler in join_handlers {
+            #[cfg(unix)]
+            if has_vdso_sgx_enter_enclave() {
+                // * The enclave thread may be in a long-running `AEX/enclu[ERESUME]` loop.
+                // Issuing a signal to return execution control back to the enclave-runner's worker thread.
+                // * In non-vdso case execution control is claimed during AEX using special handler provided by AEP address
+                // in coenter() function and signal is not required
+                unsafe { libc::pthread_kill(handler.as_pthread_t() as _, signal::SIGHUP as _); }
```

Comment:
Why is this conditional on `has_vdso_sgx_enter_enclave`?

Reply:
there are 2 different solutions for vdso/non-vdso:
with vdso: enclave execution is interrupted by a signal and RIP rewriting.
withouth vdso: enclave execution interrupted using special instruction under AEP, no signal is needed